### PR TITLE
Locales on the fly: remote locale lazy loading

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Dependencies | - https://github.com/ripe-tech/ripe-commons-pluginus/pull/267 <br> - https://github.com/ripe-tech/ripe-white/pull/981 |
| Decisions | - Add plugin to do locale bundle lazy loading from `locale_changed` event |

### CODE REVIEW

**Review by hugo-gomes — COMMENTED** [review]
@3rdvision what's the status of this?

**Comment by 3rdvision** [comment]
> @3rdvision what's the status of this?

After a meeting with @BeeMargarida and @joamag this is now awaiting an implementation to do lazy loading.
My bad, I should have updated with a comment.
Assigning to me until it's done.

**Review by hugo-gomes — COMMENTED** [review]
@3rdvision thanks! no problem :)

File: js/util/locale-bundle-loader.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    async onLocaleChanged(locale = this.locale) {
+        this._tryLoadLocaleBundle(locale);
+    }
+
+    async _tryLoadLocaleBundle(locale) {
+        // if ripeProvider isn't ready, try again later
+        if (!this.ripeProvider.ripe) {
+            const ripeReadyInterval = setInterval(async () => {
```
Comment (joamag) [inline js/util/locale-bundle-loader.js:28]: avoid busy waiting and use a promise to wait for an event triggered when ripe provider is ready

File: js/util/locale-bundle-loader.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            }, 250);
+            return;
+        }
+
+        await this.loadLocaleBundle(locale);
+    }
+
+    async loadLocaleBundle(locale) {
+        // skip if bundle is already loaded
+        if (this.loadedBundles.includes(locale)) return;
```
Comment (joamag) [inline js/util/locale-bundle-loader.js:42]: use the ripe-sdk `initBundles`

**Review by joamag — CHANGES_REQUESTED** [review]

**Pull request merged**